src/main/scala/com/stratio/receiver/RabbitMQInputDStream.scala

```diff
@@ -119,28 +119,38 @@ class RabbitMQReceiver(params: Map[String, String], storageLevel: StorageLevel)
   }
 
   def getQueueName(channel: Channel): String = {
+    // Get the queue name to use (explicit vs auto-generated).
     val queueName = checkQueueName()
-    routingKeys match {
-      case Some(routingKey) => {
-        log.info("declaring topic queue")
-        channel.exchangeDeclare(exchangeName, exchangeType, true)
 
-          val queue = channel.queueDeclare(queueName, true, false, false, new util.HashMap(0)).getQueue()
-
-        for (routingKey: String <- routingKey.split(",")) {
-          log.info("binding to routing key " + routingKey)
-          channel.queueBind(queue, exchangeName, routingKey)
+    // If no exchange was specified, then create a direct connection to the queue.
+    if(exchangeName == null || exchangeName.trim.isEmpty) {
```

Comment:
Hi Kevin, in the actual code the exchangeName never will be null. 
I suggest to do it like this:

 private val exchangeName: Option[String] = params.get("exchangeName")

 def getQueueName(channel: Channel): String = {
    // Get the queue name to use (explicit vs auto-generated).
    val queueName = checkQueueName()

    // If no exchange was specified, then create a direct connection to the queue.

      exchangeName match {
        case Some(exchangeName) => {
          log.info(s"declaring exchange '$exchangeName' of type '$exchangeType'")
          channel.exchangeDeclare(exchangeName, exchangeType, true)

          log.info("declaring topic queue")
          channel.queueDeclare(queueName, true, false, false, new util.HashMap(0))

          // Bind the exchange to the queue.
          routingKeys match {
            case Some(routingKey) => {
              // If routing keys were provided, then bind using them.
              for (routingKey: String <- routingKey.split(",")) {
                log.info(s"binding to routing key '$routingKey'")
                channel.queueBind(queueName, exchangeName, routingKey)
              }
            }
        case None =>
              log.info("binding exchange using empty routing key")
              channel.queueBind(queueName, exchangeName, "")
            }
        }
        case None => {
          log.info("exchange was not configured; declaring direct queue")
          channel.queueDeclare(queueName, true, false, false, getParams.asJava)
        }
      }

    queueName
  }